src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -34,16 +46,26 @@ internal sealed class SlidingTextWindow : IDisposable
         /// </summary>
         public const char InvalidCharacter = char.MaxValue;
 
-        private const int DefaultWindowLength = 2048;
+        /// <summary>
+        /// This number was picked based on the roslyn codebase.  If we look at parsing all files,
+        /// this window ensures that 97.4% of all characters grabbed are within the window.  Note
+        /// that this is lower than expected, but heavily influenced by how heavily roslyn uses
+        /// large strings in its codebase to represent test content.  With all these files, lexemes
+        /// average around 60 characters.  If tests are excluded, then average lexeme length drops
+        /// to 25 characters, and the hit rate goes up to 98.4%.  Because this hitrate is so high,
+        /// we don't bother to dynamically resize the window.  Instead, when we hit the end, we just
+        /// grab in the next chunk of characters from the source text, as only one in every 65 lexemes
+        /// will need to go back to the source-text to read the full lexeme.
+        /// </summary>
+        public const int DefaultWindowLength = 4096;
```

Comment:
Note: we could also just go straight up to 32768.  This is an amount that is under the LOH.  At that amount, we go to 99.8% of tokens always being read from the array directly.

Reply:
ah.  i'll go even better!  in debug it can be a non-constant, so we can actually try out a bunch of sizes and validate the core behavior is fine (including small sizes, and up to 32768).  IN release, we'll just constant this to a good size.  Then we can have best of all worlds :)